### ISSUE TITLE
Account Deactivated During Registration

### ISSUE THREAD

**Issue by tjhiggins** [vancluever/terraform-provider-acme#110]
If the acme registration fails it doesn't cleanup gracefully. The following error for a TLS handshake timeout is an example:
```
Error: get directory at 'https://acme-staging-v02.api.letsencrypt.org/directory': Get https://acme-staging-v02.api.letsencrypt.org/directory: net/http: TLS handshake timeout

  on modules/kubernetes/gateway/main.tf line 90, in resource "acme_registration" "reg":
  90: resource "acme_registration" "reg" {
```

Every subsequent apply gives the following error that the account is deactivated.
```
Error: acme: error: 403 :: POST :: https://acme-staging-v02.api.letsencrypt.org/acme/new-acct :: urn:ietf:params:acme:error:unauthorized :: An account with the provided public key exists but is deactivated, url: 

  on modules/kubernetes/gateway/main.tf line 90, in resource "acme_registration" "reg":
  90: resource "acme_registration" "reg" {
```

**Comment by vancluever** [comment]
Hey @tjhiggins, it sounds like this might be something that you'd have to try again with a new public key, unfortunately. Sounds like the registration created, but it failed on the initial re-read of the data.

Thanks!

**Comment by daraul** [comment]
Sorry, but how do I set the new public key?

**Comment by vancluever** [comment]
@daraul you will want to start with a new configuration completely, more than likely. If the public key for the registration has been de-activated and you have an error in the fashion given above, there is more than likely no way to re-activate it. You may want to ask the CA you are working with (ie: Let's Encrypt or other ACME-compliant CA).

**Comment by daraul** [comment]
I thought that simply changing the names of my resources would re-generate them all, resulting in a completely new configuration, but that didn't work. Would removing the resources from my state file before applying work?

**Comment by vancluever** [comment]
@daraul for this specific issue, you would need to destroy the public key too, just keep that in mind if you didn't do that as well. Registrations are tied to the public key, so if the registration tied to the key is deactivated, that key is off limits.

**Comment by daraul** [comment]
> if the registration tied to the key is deactivated, that key is off limits.

I figured as much, I couldn't find a way to deactivate the key, or destroy it. 

I think i got lucky (or unlucky) in that i had forgotten that I'd configured the provider with acme's staging server. My troubles started when I just switched the endpoint to production and attempted to apply. 

I suppose it's possible that public key exists on the staging server still, but will that be a problem in the long run?

**Comment by vancluever** [comment]
Yeah that's weird, if you were using `tls_private_key` like in the examples, destroying the resource should have done it. If you were bringing in the key from an outside source, of course that changes in that you would need to make sure the key was rotated in whatever OOB process you were using.

Endpoint changes are something that would be nice to try and fix eventually, as it's a common source of confusion. I'm always torn between adding something and not though, as it's usually something that gets fixed via a workspace change (new state) and as such doesn't persist, and prevents unintended consequences from changing the endpoint of a production workspace.

> I suppose it's possible that public key exists on the staging server still, but will that be a problem in the long run?

Public keys are public for a reason 🙂 I think the worst of it would be that you would not be able to use that key for accounts on the staging server, but that's it.

**Comment by daraul** [comment]
> if you were using tls_private_key like in the examples, destroying the resource should have done it

I will confess I didn't follow the instructions to the letter until I noticed something was wrong, so that's my bad 😄
It's possible I messed up my state by changing the names of the resources in an attempt to just replace them all in one go. 

> unintended consequences from changing the endpoint of a production workspace.

My knee jerk response would be to leave things as they are. Particularly in my case, I am almost definitely at fault here. Changing things so that I get off the hook for not paying attention isn't something I would do if I were you lol
